identity/src/rsa.rs

```diff
@@ -109,7 +156,7 @@ impl PublicKey {
 
     /// Decode an RSA public key from a DER-encoded X.509 SubjectPublicKeyInfo
     /// structure. See also `encode_x509`.
-    pub fn decode_x509(pk: &[u8]) -> Result<PublicKey, DecodingError> {
+    pub fn try_decode_x509(pk: &[u8]) -> Result<PublicKey, DecodingError> {
```

Comment:
Here is an idea:

Given that we have a standard encoding in libp2p for each keytype, I think it would make sense to just call these `try_decode` on each key pair. That would be more discoverable and there isn't really a need to "qualify" the function with information from the encoding because there is only one encoding.

What do you think?

Reply:
That makes sense.

Reply:
No actually I think we need a way to import keys from outside of `libp2p` ecosystem, and I guess that's why the method is here.